File: TrabajoFinalPrueba1/PegasusV1/PegasusV1/Entities/DocenteMateria.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace PegasusV1.Entities
{
    [Table("DOCENTE_MATERIA")]
    public class DocenteMateria
    {
        public int Id { get; set; }

        [ForeignKey("Id_Docente")]
        public Usuario? Docente { get; set; }
        public int? Id_Docente { get; set; }

        [ForeignKey("Id_Materia")]
        public Materia? Materia { get; set; }
        public int? Id_Materia { get; set; }

        [ForeignKey("Id_Curso")]
        public Curso? Curso { get; set; }
        public int? Id_Curso { get; set; }
    }
}

File: pegasusapp/src/models/docenteMateria.ts
import { Materia } from './materia';
import { Usuario } from './usuario';
import { Curso } from './curso';

export interface DocenteMateria {
  id: number;
  id_Materia?: number | null;
  id_Docente?: number | null;
  id_Curso?: number | null;

  materia?: Materia | null;
  docente?: Usuario | null;
  curso?: Curso | null;
}
